scripts/format_arpa_lm.py

```diff
@@ -41,13 +42,45 @@
     sys.exit("format_arpa_lm.py: expected directory {0} to exist.".format(
             args.temp_dir))
 
+# verify the input string max_memory
+if args.max_memory != '':
+    # valid string max_memory must have at least two items 
+    if len(args.max_memory) >= 2:
+        s = args.max_memory
+        # case1: valid string max_memory can be formatted as "integer + letter or '%'"
+        # the unit of memory size can also be 'T', 'P', 'E', 'Z', or 'Y'. They
+        # are not included here considering their rare use in practice
+        if s[-1] in ['b', '%', 'K', 'M', 'G']:
+            if not s[-2].isdigit():
```

Comment:
Good, I didn't find any other issue. You could do some regression tests and then we can merge it.